deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -147,11 +148,13 @@ def run(self, train_dataset: Dataset, test_dataset: Dataset, model=None) -> Chec
 
         # Create scoring function, used to calculate the per sample model error
         if task_type == ModelType.REGRESSION:
-            def scoring_func(dataset):
+            def scoring_func(dataset: Dataset):
                 return per_sample_mse(dataset.label_col, model.predict(dataset.features_columns))
         else:
-            def scoring_func(dataset):
-                return per_sample_binary_cross_entropy(dataset.label_col,
+            def scoring_func(dataset: Dataset):
+                le = preprocessing.LabelEncoder()
+                encoded_label = le.fit_transform(dataset.label_col)
+                return per_sample_binary_cross_entropy(encoded_label,
```

Comment:
Why not do the encoding outside of the function? I think that the implicit assumption that LabelEncoder.fit_transform will give the same result for train and test might cause bugs in some future.

Reply:
did a fix